File: cpp_roadmap/vet_clinic/filing/src/pet.cpp
#include <pet.h>

// super simple stuff, but could be more complex in the future

Pet::Pet(std::string name, PetType pet_type, Sex sex, double weight, IDManager& id_manager) : name_(name), pet_type_(pet_type), sex_(sex), weight_(weight) {
    id_ = id_manager.get_new_id();
}

std::string Pet::get_pet_type_string() {
    switch(pet_type_){
        case PetType::CAT: {
            return "cat";
        } 
        case PetType::DOG: {
            return "dog";
        } 
        case PetType::RABBIT: {
            return "rabbit";
        } 
        case PetType::SNAKE: {
            return "snake";
        } 
        case PetType::LIZARD: {
            return "lizard";
        } 
        default: {
            return "Unknown";
        }
    }
}

std::string Pet::get_sex_string() {
    switch(sex_){
        case Sex::MALE: {
            return "male";
        } 
        case Sex::FEMALE: {
            return "female";
        } 
        default: {
            return "Unknown";
        }
    }
}

void Pet::change_name(std::string new_name) {
    name_ = new_name;
}

void Pet::update_pet_weight(double weight) {
    weight_ = weight;
}

File: dotnet_vet_clinic/Filing/Models/Pet.cs
namespace Filing.Models;

public class Pet(string chipNumber, string name, Pet.Type petType, double weight)
{
    // TODO get this type to be inserted as string
    public enum Type
    {
        Unknown = 0,
        Cat = 1,
        Dog = 2,
        Rabbit = 3,
        Snake = 4,
        Lizard = 5
    }


    public string ChipNumber { get; } = chipNumber;
    public string Name { get; set; } = name;
    public Type PetType { get; set; } = petType;
    public double Weight { get; set; } = weight;
}